### DISCUSSION TITLE
How to avoid convergence error

### DISCUSSION THREAD

**Discussion by cgs1990** [e2nIEE/pandapower#1287] [Q&A]
Hello everybody!

I have a usage question, is it possible to get if an OPF converges or not without failing in python? That is, to say whether it converges or not, simply. 

For example, it is the case of a loop, in which I set the parameters of a line, depending on the parameter the problem converges or not. If the parameter is optimized, there may be parameters for which the problem converges or not, but I need it to simply say it does not converge, by means of a signal, since if it shows error in python it disables me from being able to optimize anything.

**Comment by rbolgaryn** [comment]
Dear @cgs1990,

you can get around it by simply putting the OPF calculation in the try...except block and catching the Exception for when OPF did not converge (you need to select a specific Exception so that it is selective).

Best regards, 
Roman
